libraries/tock-register-interface/Cargo.toml

```diff
@@ -13,3 +13,6 @@ edition = "2018"
 
 [badges]
 travis-ci = { repository = "tock/tock", branch = "master" }
+
+[features]
+std_unit_tests = []
```

Comment:
That's my personal opinion, but tests should be enabled by default (either by putting this feature in the `default` group or by renaming it as the opposite of this feature). Not generating the tests should be an opt-in action.

Reply:
I can understand both angles to this and would be fine either way. Like mentioned in the PR, my stance on it was that tock-registers is explicitly `no_std` by default, not opt-in `no_std`, so it is already in use out there in no_std crates.

So if we would go the std-by-default way, I would at least also make the lib.rs global no_std attribute optional accordingly. Otherwise, we have a mix-up.

Let's hear what the code oweners have to to say about this :)

Reply:
Makes sense from the perspective of external crates depending on this. From Tock's side, I see that this crate is depended on in 3 places:

```
$ git grep tock-registers
arch/rv32i/Cargo.toml:tock-registers = { path = "../../libraries/tock-register-interface" }
kernel/Cargo.toml:tock-registers = { path = "../libraries/tock-register-interface" }
libraries/riscv-csr/Cargo.toml:tock-registers = { path = "../tock-register-interface" }
```

So having the tests generated as an opt-in risks that one of the (existing or future) dependencies accidentally doesn't generate them. Note that tests running as `std` are still in discussion in https://github.com/tock/tock/pull/1393.